Include/internal/pycore_frame.h

```diff
@@ -31,6 +31,7 @@ typedef struct _interpreter_frame {
     int f_lasti;       /* Last instruction if called */
     int stacktop;     /* Offset of TOS from localsplus  */
     PyFrameState f_state;  /* What state the frame is in */
+    int depth; /* Depth of the frame in a ceval loop */
```

Comment:
The `depth` is on the frame object and not on the `cframe` because this is **much** easier to fetch from gdb and other debugging tools.

Reply:
Why is `tstate->cframe->depth` any harder than `tstate->frame->depth`?

Reply:
Because is not `tstate->cframe->depth` or `tstate->frame->depth` what gdb fetches. `gdb` gets the frame from a local variable from the stack so you just start with the frame pointer.

Also, if you place it in `tstate->cframe->depth` now you need to fetch *both* (`cframe` and `frame`) and match them accordingly, which is much more complicated.